source/conditional-keys.d.ts

```diff
@@ -36,10 +36,16 @@ export type ConditionalKeys<Base, Condition> =
 	[Key in keyof Base]-?:
 	// Pick only keys with types extending the given `Condition` type.
 	Base[Key] extends Condition
-	// Retain this key since the condition passes.
-		? Key
-	// Discard this key since the condition fails.
+		// Retain this key since the condition passes
+		?
+			// If the condition extends never, only include the key if the value for that key is also never
```

Comment:
let me know if you want to format this differently. I chose to progressively increase indentation for each level of ternary, and to keep comments indented to the same level as the line they're describing.